core/src/main/java/io/questdb/griffin/engine/functions/json/JsonPathPrimitiveFunction.java

```diff
@@ -0,0 +1,373 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.json;
+
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.cairo.sql.ScalarFunction;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.std.BinarySequence;
+import io.questdb.std.Long256;
+import io.questdb.std.json.SimdJsonError;
+import io.questdb.std.json.SimdJsonResult;
+import io.questdb.std.str.*;
+import org.jetbrains.annotations.Nullable;
+
+public class JsonPathPrimitiveFunction implements ScalarFunction, BinaryFunction, JsonPathFunction {
```

Comment:
This function should override `isReadThreadSafe` since otherwise a single instance of this function may be called from multiple threads:
```java
    @Override
    public boolean isReadThreadSafe() {
        return false;
    }
```

I didn't check other new functions here, so there is a chance `isReadThreadSafe` should be overriden there too.

Reply:
Also, would be great to add a few tests around this function to `ParallelFilterTest` and `ParallelGroupByFuzzTest`.

Reply:
i am on it